apps/designer/components/Layout.tsx

```diff
@@ -11,42 +11,47 @@ import SavePopup from "./SavePopup";
 const Layout: React.FC<ScriptProps> = ({ children }) => {
   // TODO: improve this type later
   const {
-    myNavbar,
-    isActive,
-    welcomeClick,
-    openWelcome,
-    handleClick,
-    toggleButton,
-    myToggleButton,
-    newButtonClick,
-    clickSaveButton,
-    closeSaveButton,
-    openPopup,
+    isNavbarExpand,
+    isWelcomePanelActive,
+    handleClickOnWelcome,
+    isWelcomeComponentOpen,
+    handleClickOnNewDesign,
+    isToggled,
+    handleClickForToggle,
+    handleClickOnNewButton,
+    handleClickOnSaveButton,
+    closeSavePopup,
+    isSavePopupOpen,
   } = navigationUseStore();
 
   return (
     <>
       <Navbar
-        myNavbar={myNavbar}
-        toggleButton={toggleButton}
-        myToggleButton={myToggleButton}
-        newButtonClick={newButtonClick}
-        clickSaveButton={clickSaveButton}
+        isNavbarExpand={isNavbarExpand}
+        isToggled={isToggled}
+        handleClickForToggle={handleClickForToggle}
+        handleClickOnNewButton={handleClickOnNewButton}
+        handleClickOnSaveButton={handleClickOnSaveButton}
       />
       {children}
-      {openWelcome === false && <NewDesign handleClick={handleClick} />}
-      {openWelcome === true && (
+      {isWelcomeComponentOpen === false && (
+        <NewDesign handleClickOnNewDesign={handleClickOnNewDesign} />
+      )}
+      {isWelcomeComponentOpen === true && (
         <Welcome
-          welcomeClick={welcomeClick}
-          isActive={isActive}
-          myToggleButton={myToggleButton}
-          toggleButton={toggleButton}
+          handleClickOnWelcome={handleClickOnWelcome}
+          isWelcomePanelActive={isWelcomePanelActive}
+          handleClickForToggle={handleClickForToggle}
+          isToggled={isToggled}
         />
       )}
-      {openPopup === true && (
-        <SavePopup closeSaveButton={closeSaveButton} openPopup={openPopup} />
+      {isSavePopupOpen === true && (
+        <SavePopup
+          closeSavePopup={closeSavePopup}
+          isSavePopupOpen={isSavePopupOpen}
+        />
       )}
-      {isActive === false && <LoadingBar />}
+      {isWelcomePanelActive === false && <LoadingBar />}
     </>
   );
 };
```

Comment:
We do not need to do further refactoring of this page currently. But this page seems to require the use of state machine based approach using `xstate` library. We will do this refactor when we add more complexity to these pannels and layouts. @NamanRajputXXI @abhishekrathi16

Reply:
If not `xstate` it should atleast be refactored to use a ts union type instead of a bunch of related booleans.